lib/possible_bipartition.rb

```diff
@@ -1,4 +1,43 @@
+# I looked at the leetcode solution in Python 
+# (https://leetcode.com/problems/possible-bipartition/solution/) 
+# to get an idea of how to turn the list of dislikes into a graph, 
+# but chose to implement a breadth-first solution instead
 
 def possible_bipartition(dislikes)
-  raise NotImplementedError, "possible_bipartition isn't implemented yet"
+    return true if dislikes.empty?
+    graph = build_graph(dislikes)
+    # start with an arbitrary node of the graph, 
+    # assuming all nodes have at least one edge
+    grouped = {graph.keys[0] => true}
+    queue = [graph.keys[0]]   
+    # using a bfs approach, set all neighbor nodes to the opposite label
+    # and add them to the queue if they haven't been visited
+    # return false if a neighbor has the same label as the current node
+    while !queue.empty?
+      node = queue.shift()
+      graph[node].each do |neighbor|
+        if grouped[neighbor] == nil
+          grouped[neighbor] = !grouped[node]
+          queue.push(neighbor)
+        elsif grouped[neighbor] == grouped[node]
+          return false
+        end  
+      end     
+      if queue.empty? && node < graph.keys.max
+        queue.push(node + 1)
+      end
```

Comment:
This doesn't seem like the most efficient way to handle a disconnected graph.  Maybe also check to ensure that `grouped[node].nil?`

Reply:
I updated this to seek the next ungrouped node. It gets rid of a lot of the aimless wandering of the last approach, but still doesn't seem very efficient. Thoughts?:
```
if queue.empty? 
    graph.keys.each do |key|
       if grouped[key].nil?
           queue.push(key)
           break
       end
    end
end
```